.circleci/config.yml

```diff
@@ -495,6 +508,17 @@ workflows:
               cniNetworkPlugin: [ calico ]
           parallelism: 1
           requires: [ build-<< matrix.arch >>, check ]
+      - e2e:
+          name: << matrix.target >>:<< matrix.arch >>-<< matrix.k8sVersion >>-ebpf
+          matrix:
+            alias: test/e2e-legacy-ebpf
+            parameters:
+              k8sVersion: [ kind ]
+              target: [ test/e2e ]
+              arch: [ amd64 ]
+              ebpf: [ enabled ]
+          parallelism: 1
+          requires: [ build-<< matrix.arch >>, check ]
```

Comment:
eBPF doesn't work with K3D, and we skip kind for most of the tests. `eBPF` is used to skip legacy tests which not supposed to run with kind.

Reply:
It does run on legacy when using ipv6 see: https://app.circleci.com/pipelines/github/kumahq/kuma/16887/workflows/287ee844-230a-40d7-8c14-0b0f51a616aa/jobs/237169 . 

Should we relax running kind with legacy on v4?

BTW why is eBPF not working on k3d?

Reply:
Also if this test matrix stuff is not viable should we look at alternatives?

Reply:
Feels like maybe we should use: https://circleci.com/blog/building-cicd-pipelines-using-dynamic-config/

Reply:
We use kind for ipv6 but I think we disabled it for ipv4 because it was flaky/or just because we are using K3D. With K3D I had problems with `/sys/fs/bpf` not being shared, I could probably make it shared in some tricky way but thought using kind might be more predictable in this case.

Reply:
Is there a k3d issue to track this?

Reply:
I've changed to K3d with volume mounting and now it should work.

Reply:
There is an issue that causes test to fail on circleci but not on the local machine. ```warning: found mount /sys/fs/cgroup, but it's type (tmpfs) is not cgroup2 - ignoring
Error: failed to setup transparent proxy: getting cgroup failed with error: cannot find any cgroup2 mounts``` I am looking for the best way to solve it but so far kind is the easiest

Reply:
Ok k3d is a bit of a pain: no ipv6, no ebpf... Should we just go back to kind and not use k3d at all?

Reply:
Looks like so far everything works. Needs to merge things from https://github.com/kumahq/kuma/pull/5176